tensorflow_quantum/python/differentiators/gradient_test.py

```diff
@@ -220,5 +236,97 @@ def test_analytic_value_with_simple_circuit(self, differentiator, op):
+            expectations = op(circuit, tf.convert_to_tensor(['alpha']),
+                              symbol_values_tensor, psums,
+                              tf.convert_to_tensor([[num_samples]]))
+        grads = g.gradient(expectations, symbol_values_tensor)
+        ground_truth_grads = np.array([[-1.1839752]])
+        self.assertAllClose(ground_truth_grads, grads, rtol=0.2, atol=0.2)
+
+    @parameterized.parameters(
+        list(
+            util.kwargs_cartesian_product(
+                **{
+                    'diff_and_tol': zip(SAMPLED_DIFFS, SAMPLED_DIFFS_TOLS),
+                    'op': SAMPLED_OPS,
+                    'n_qubits': [3],
+                    'n_programs': [5],
+                    'n_ops': [2],
+                    'symbol_names': [['a', 'b']],
+                    'num_samples': [30000]
+                })))
+    def test_approx_equality_shallow(self, diff_and_tol, op, n_qubits,
+                                     symbol_names, n_ops, n_programs,
+                                     num_samples):
+        """Test small circuits with limited depth."""
+        differentiator, tol = diff_and_tol
+        differentiator.refresh()
+        op = differentiator.generate_differentiable_op(sampled_op=op)
+
+        qubits = cirq.GridQubit.rect(1, n_qubits)
+        circuit_batch, resolver_batch = \
+            util.random_symbol_circuit_resolver_batch(
+                cirq.GridQubit.rect(1, n_qubits), symbol_names, n_programs)
+
+        # Prepare random pauli sums and add initial superposition gates.
+        psums = []
+        for i in range(len(circuit_batch)):
+            psums.append(util.random_pauli_sums(qubits, 1, n_ops))
+            circuit_batch[i] = cirq.Circuit(
+                cirq.H.on_each(qubits)) + circuit_batch[i]
+
+        symbol_values_array = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names]
+             for resolver in resolver_batch],
+            dtype=np.float32)
+
+        # calculate tfq gradient
+        symbol_values_tensor = tf.convert_to_tensor(symbol_values_array)
+        programs = util.convert_to_tensor(circuit_batch)
+        ops = util.convert_to_tensor(psums)
+        with tf.GradientTape() as g:
+            g.watch(symbol_values_tensor)
+            expectations = op(
+                programs, tf.convert_to_tensor(symbol_names),
+                symbol_values_tensor, ops,
+                tf.convert_to_tensor([[num_samples] * n_ops] * n_programs))
+        tfq_grads = g.gradient(expectations, symbol_values_tensor)
+
+        # calculate gradients in cirq using a very simple forward differencing
+        # scheme
+        cirq_grads = _cirq_simple_finite_difference(circuit_batch,
+                                                    resolver_batch,
+                                                    symbol_names, psums)
+
+        self.assertAllClose(cirq_grads, tfq_grads, rtol=tol, atol=tol)
```

Comment:
I think `rtol` should be removed, for random circuits gradients could get near zero and then `rtol` might be flagging things unnecessarily